From d2ab43eca7d3b9cb28520e333d499bafbfeef751 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Thu, 20 Aug 2015 01:11:42 +0800
Subject: [PATCH 1020/1221] MLK-11391 ARM: imx: make sure SNVS clk is enabled
 during suspend

As SNVS clk may be disabled in kernel to save power(~1mA),
but during suspend/resume, we have to access SNVS register
to do MMDC retention and power down SoC etc.., need to
make sure SNVS clk is enabled before accessing its register.

Signed-off-by: Anson Huang <b20788@freescale.com>
[zou:Original patch taken from
git.freescale.com/imx/fsl-arm-yocto-bsp.git-b imx-3.14.52-1.1.0_ga]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 arch/arm/mach-imx/suspend-imx7.S |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-imx/suspend-imx7.S b/arch/arm/mach-imx/suspend-imx7.S
index d8da34c..885008c 100644
--- a/arch/arm/mach-imx/suspend-imx7.S
+++ b/arch/arm/mach-imx/suspend-imx7.S
@@ -88,6 +88,8 @@
 #define DDRC_SWSTAT	0x324
 #define DDRPHY_LP_CON0	0x18
 
+#define CCM_SNVS_LPCG	0x250
+
 	.align 3
 
 	.macro	disable_l1_dcache
@@ -452,6 +454,12 @@
 ENTRY(imx7_suspend)
 	push	{r4-r12}
 
+	/* make sure SNVS clk is enabled */
+	ldr	r11, [r0, #PM_INFO_MX7_CCM_V_OFFSET]
+	add	r11, r11, #0x4000
+	ldr	r7, =0x3
+	str	r7, [r11, #CCM_SNVS_LPCG]
+
 	/* check whether it is a standby mode */
 	ldr	r11, [r0, #PM_INFO_MX7_GPC_V_OFFSET]
 	ldr	r7, [r11, #GPC_PGC_C0]
-- 
1.7.5.4

